test/integration/scheduler/priorities_test.go

```diff
@@ -496,3 +496,75 @@ func TestDefaultPodTopologySpreadScore(t *testing.T) {
 		})
 	}
 }
+
+// TestDefaultSchedulerTolerationInfluence tests if the increased weight of Toleration priority(which is default) has
+// an influence on the pod placement
+// The algorithm is pretty simple.
+// 1.) Start the scheduler with default config.
+// 2.) Create 3 nodes with one of them having taint.
+// 3.) Create 80 pods with with no tolerations. (80 is chosen randomly with the qps limit of 100(?) in mind)
+// 4.) < 30% of the pods should land on the tainted node.
+func TestDefaultSchedulerTolerationInfluence(t *testing.T) {
+	testCtx := initTest(t, "")
+	defer testutils.CleanupTest(t, testCtx)
+	cs := testCtx.ClientSet
+	var taintedNodes []*v1.Node
+
+	// Add a few nodes with no taints
+	_, err := createAndWaitForNodesInCache(testCtx, "zonenode1", st.MakeNode().Label(v1.LabelTopologyZone, "z1"), 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = createAndWaitForNodesInCache(testCtx, "zonenode2", st.MakeNode().Label(v1.LabelTopologyZone, "z2"), 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add tainted node
+	taintedNodes, err = createAndWaitForNodesInCache(testCtx, "zonenode3", st.MakeNode().Label(v1.LabelTopologyZone, "z3").Taint("newtaint", "newvalue", true), 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ns := testCtx.NS.Name
+	var pods []v1.Pod
+	pause := imageutils.GetPauseImageName()
+	totalPodCount := 80
+	for i := 0; i < totalPodCount; i++ {
+		p := st.MakePod().Name(fmt.Sprintf("p-%d", i)).Container(pause).Obj()
+		_, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, p, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Cannot create Pod: %v", err)
+		}
+	}
+	// Wait for all Pods to be scheduled.
+	err = wait.Poll(pollInterval, wait.ForeverTestTimeout, func() (bool, error) {
+		podList, err := cs.CoreV1().Pods(ns).List(testCtx.Ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Cannot list pods to verify scheduling: %v", err)
+		}
+		for _, p := range podList.Items {
+			if p.Spec.NodeName == "" {
+				return false, nil
+			}
+		}
+		pods = podList.Items
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("All the pods did not get scheduled: %v", err)
+	}
+	podsOnTaintedNode := 0
+	for _, pod := range pods {
+		if pod.Spec.NodeName == taintedNodes[0].Name {
+			podsOnTaintedNode++
+		}
+	}
+	podsOnTaintedNodePercentange := float64(podsOnTaintedNode) / float64(totalPodCount)
+	if podsOnTaintedNodePercentange > 0.3 {
+		// Why 30%? Scheduler when impartial should allow 1/3rd of the pods to land on the taintedNode but with taint
```

Comment:
30% is not too far from 1/3.

Is that all it gets? You run this one with v1beta2, right?

Reply:
Yes, I wanted to make sure that it is less than default value of 1/3rd. Locally it runs at 20% or lower but I don't want to flake as it is best effort. I can keep it 25% to be on safe side. Let me know what you think of it.

Reply:
are those percentages for the old weights or new weights?

Reply:
New weights. That's why I included all the commits from https://github.com/kubernetes/kubernetes/pull/104251/ and you can see the score of 300 [here](https://github.com/kubernetes/kubernetes/pull/105386#issuecomment-931741812)

Reply:
Can you tell the difference from old weights?

Reply:
Is the goal to test how new weight co-work with **all** other score plugins? or, we can just selectively enable a couple of score plugins to make the behavior more deterministic.

Reply:
Co-work with all other plugins. IOW, guage the impact of this increased weight in default scheduler.

Reply:
>  Can you tell the difference from old weights?

It was 100 but I am rerunning the test. Will post the status in a bit.

Reply:
Yes, it was 100:

```
I1005 16:10:27.126202   10654 generic_scheduler.go:434] "Plugin scored node for pod" pod="9bb35d39-143c-4260-93fc-b4df2a09c571/p-1" plugin="TaintToleration" node="zonenode2-0" score=100
```

Reply:
Sorry, my question was about the percentage of pods that were scheduled in the tainted node. In other words, what is the delta effect of the new weights in the behavior.

Reply:
The delta between scores of the nodes(without and with increased weights) is the difference here. 


```
With weight of 1:

I1006 09:30:38.827663   21434 generic_scheduler.go:491] "Calculated node's final score for pod" pod="c0f0a078-dd79-4567-b0cf-ef80626ed713/p-0" node="zonenode3-0" score=300
I1006 09:30:38.827672   21434 generic_scheduler.go:491] "Calculated node's final score for pod" pod="c0f0a078-dd79-4567-b0cf-ef80626ed713/p-0" node="zonenode2-0" score=400




With increased weights:
I1006 09:38:19.607151   27020 generic_scheduler.go:491] "Calculated node's final score for pod" pod="50f30f2e-a27c-47ed-aa0a-b546b51df011/p-0" node="zonenode2-0" score=600
I1006 09:38:19.607161   27020 generic_scheduler.go:491] "Calculated node's final score for pod" pod="50f30f2e-a27c-47ed-aa0a-b546b51df011/p-0" node="zonenode3-0" score=300
```

You can notice that the nodes

The result is sometimes(since this is best effort), we see that the value of `podsOnTaintedNodePercentange` on tainted node goes beyond 20%.

Reply:
So I'm interested to know what is the average value of `podsOnTaintedNodePercentange` for original weights and the new weights.

> The scheduler behavior wouldn't change by much as this is best effort.

Then what is the point of changing the weights? :)

I'm sure there is an improvement. I just want it properly documented for future reference.

Reply:
My 2 cents is that the expected `podsOnTaintedNodePercentange` (30%) is a bit conservative and thus it cannot quite verify if the new weight work or not. If the behavior is too random to predict, we either shouldn't build this test, or just selectively enable a couple of score plugins to make the behavior deterministic.

Reply:
Sorry @Huang-Wei just saw your msg

> podsOnTaintedNodePercentange (30%) is a bit conservative and thus it cannot quite verify if the new weight work or not.

I agree, the local runs have always been in the <=20% range. If the folks are fine keeping it for 20% range for now and revisit it when the test flakes, I am fine with the change.

I am also thinking of introducing another test which will:
- Create nodes with taint
- Create pods with soft affinity
- With the increased weight of `TaintToleration` plugin, most of the pods should land onto a node without a taint as affinities have a weight of 2 vs `TaintToleration` have 3.
This quantifies the change in behavior in another dimension

The goal when I started working on this feature is to ensure that the `TaintToleration` have higher weight so that most pods would land onto untainted node which I am able to prove(in a best effort way). This is needed for us when we do upgrades in OpenShift. I can also quantify behavioral changes from multiple dimensions. I want to know which
dimensions the sig is interested in.

Reply:
20% (pods spread as 2:2:1) seems to be a reasonable value that is able to verify the new weight.

Reply:
An improvement of 5% doesn't sound like much considering that you tripled the weight.

However, I question the usefulness of the test as it is now. There are actually a couple of hidden things in the test setup: what is the node allocatable and what are the pod requests? Setting appropriate values to match reasonable real world scenarios would give us a better perspective of the improvement.

> how do you want this to be documented?

It could just be part of the PR description, or we could also include it in the release note.

Reply:
What I'm saying is that in the test is quite hidden what is the size of the node and the requests of the pods. Can you add explicit numbers? I guess something reasonable is 8 core VMs with pods requesting 250mcpu